pkg/kv/kvserver/spanset/batch.go

```diff
@@ -897,9 +980,9 @@ func DisableReaderAssertions(reader storage.Reader) storage.Reader {
 func DisableReadWriterAssertions(rw storage.ReadWriter) storage.ReadWriter {
 	switch v := rw.(type) {
 	case ReadWriter:
-		return DisableReadWriterAssertions(v.w.(storage.ReadWriter))
+		return DisableReadWriterAssertions(v.spanSetWriter.w.(storage.ReadWriter))
```

Comment:
Hm, does this work? `v.spanSetWriter.w` is a `storage.Writer`, and we're converting it to `storage.ReadWriter`. Is Go somehow able to up-cast this interface?

Reply:
Interesting, seems to work: https://go.dev/play/p/_5jPANdw17p